### ISSUE TITLE
RadDataForms date picker crash on android

### ISSUE THREAD

**Issue by mhmo91** [telerik/nativescript-ui-feedback#871]
### Did you verify this is a real problem by searching the [NativeScript Forum](https://discourse.nativescript.org/)?
YES

### Which platform(s) does your issue occur on?
Android

### Please provide the following version numbers that your issue occurs with:
- **Progress NativeScript UI version: ``` "nativescript-ui-dataform": "^3.7.3" ``` 
- **CLI:** ``` 5.0.0-2018-10-09-12426 ``` 
- **Cross-platform modules:** ``` "tns-core-modules": "^4.3.0-2018-10-09-152728-02" ```
- **Runtime(s):** ```  "tns-android": {
      "version": "5.0.0-2018-10-09-135927-01"
    },
    "tns-ios": {
      "version": "5.0.0-2018-10-08-180008-01"
    }```
### Please tell us how to recreate the issue in as much detail as possible.
for the Dataform component, when using Date editor, if i'm initializing the sourceModel with a value '2018-10-9' and then the user tries to change it, the app crashes with this error
```
An uncaught Exception occurred on "main" thread.
java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Long
	at com.telerik.widget.dataform.engine.StringToDateConverter.convertTo(StringToDateConverter.java:38)
	at com.telerik.widget.dataform.visualization.core.EntityPropertyViewer.convertedEditorValue(EntityPropertyViewer.java:440)
	at com.telerik.widget.dataform.visualization.core.EntityPropertyEditor.onEditorValueChanged(EntityPropertyEditor.java:267)
	at com.telerik.widget.dataform.visualization.editors.DataFormDateTimeEditor.onPickerValueChanged(DataFormDateTimeEditor.java:175)
	at com.telerik.widget.dataform.visualization.editors.DataFormDateEditor.onDateSet(DataFormDateEditor.java:63)
	at android.app.DatePickerDialog.onClick(DatePickerDialog.java:171)
	at com.android.internal.app.AlertController$ButtonHandler.handleMessage(AlertController.java:168)
	at android.os.Handler.dispatchMessage(Handler.java:108)
	at android.os.Looper.loop(Looper.java:166)
	at android.app.ActivityThread.main(ActivityThread.java:7425)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:245)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:921)
```
   

### Is there code involved? If so, please share the minimal amount of code needed to recreate the problem.

for example the view
```html
 <RadDataForm #filterForm [source]="filterFormModel" validationMode="Immediate" commitMode="Immediate">
           

            <TKEntityProperty tkDataFormProperty name="dateFrom" displayName="{{'Date from' | L}}" index="1">
                <TKPropertyEditor tkEntityPropertyEditor type="DatePicker"></TKPropertyEditor>
                <TKNonEmptyValidator tkEntityPropertyValidators errorMessage="{{'Required' | L}}"></TKNonEmptyValidator>
            </TKEntityProperty>

            <TKEntityProperty tkDataFormProperty name="dateTo" displayName="{{'Date to' | L}}" index="2">
                <TKPropertyEditor tkEntityPropertyEditor type="DatePicker"></TKPropertyEditor>
                <TKNonEmptyValidator tkEntityPropertyValidators errorMessage="{{'Required' | L}}"></TKNonEmptyValidator>
            </TKEntityProperty>
        </RadDataForm>
```

in the component class for ex.
```js
            this.filterFormModel = new FilterFormModel("2018-10-08", "2018-10-30");
```

this will work fine on the IOS n as far as i remember, it was working fine on android..

Everytime i get with the client about releasing the app, something just breaks n i have to waste lots of energy and time getting around, for this issue, i have over 30 forms in the app, all of them have date picker...  please give this a priority!!!

**Comment by mhmo91** [comment]
Downgrading from ```"nativescript-ui-dataform": "^3.7.3"``` to ```"nativescript-ui-dataform": "^3.7.2"``` fixed the issue.

Guys, I urge u to make NS UI open source, and make sure your releases are stable n well tested

**Comment by mhmo91** [comment]
@tylerablake try removing the node_modules n platforms folders and then run ```npm i``` then ```tns run android --bundle```

**Comment by tylerablake** [comment]
I was able to get it working, must have been missing a step and it was caching the version or something, thanks! :)

**Comment by bjoernWahle** [comment]
Any ETA on when this is getting released?

**Comment by zbranzov** [comment]
The fix is released in nativescript-ui-dataform@3.8.
